### PR TITLE
Add timeout for SPI OAUTH route creation

### PR DESCRIPTION
**Pull request by flacatus**
In CI we need to wait few seconds until the routes are binding in redhat-appstudio workspace

### CODE REVIEW

File: hack/preview.sh [hunk 1]
```diff
@@ -67,7 +67,7 @@ echo "start spi config"
 CLUSTER_URL_HOST=$(oc whoami --kubeconfig ${CLUSTER_KUBECONFIG} --show-console | sed 's|https://console-openshift-console.apps.||')
 if ! oc get namespace spi-system --kubeconfig ${KCP_KUBECONFIG} &>/dev/null; then
   oc create namespace spi-system --kubeconfig ${KCP_KUBECONFIG}
-  oc create route edge -n spi-system --service spi-oauth-service --port 8000 spi-oauth --kubeconfig ${KCP_KUBECONFIG}
+  timeout 2m bash -x -c -- 'while ! oc create route edge -n spi-system --service spi-oauth-service --port 8000 spi-oauth --kubeconfig ${KCP_KUBECONFIG}; do sleep 1; done'
```
Comment (Michkov) [inline hack/preview.sh:70]: rather wait for host binding:
```
oc create route edge -n spi-system --service spi-oauth-service --port 8000 spi-oauth --kubeconfig ${KCP_KUBECONFIG}
timeout 2m bash -x -c -- 'while [ "$(oc get -o jsonpath='{.status.ingress[0].host}' -n spi-system route spi-oauth)" == "" ]; do sleep 1; done'
```

Comment (flacatus) [reply hack/preview.sh:70]: The problem for binding is that is saying phase Bounded but routes are not presents yet in the ws

Comment (Michkov) [reply hack/preview.sh:70]: this is running inside WS (KCP) so when the `.status.ingress[0].host` is set then it's firstly properly set inside workload cluster and synced also to KCP WS.

Comment (Michkov) [reply hack/preview.sh:70]: My testing with our code:
```
$ oc delete namespace test; oc create namespace test; timeout 2m bash -x -c -- 'while ! oc create route edge -n test --service spi-oauth-service --port 8000 test; do sleep 1; done'; oc get -n test -o yaml route test
namespace "test" deleted
namespace/test created
+ oc create route edge -n test --service spi-oauth-service --port 8000 test
route.route.openshift.io/test created
apiVersion: route.openshift.io/v1
kind: Route
metadata:
  annotations:
    kcp.dev/cluster: root:users:zu:yc:kcp-admin:redhat-appstudio
  creationTimestamp: "2022-10-20T09:03:27Z"
  finalizers:
  - workload.kcp.dev/syncer-cE1LtxcEXYvw2nQIA0qEtQJuYhiFvsguQ0eE5R
  generation: 1
  labels:
    state.workload.kcp.dev/cE1LtxcEXYvw2nQIA0qEtQJuYhiFvsguQ0eE5R: Sync
  name: test
  namespace: test
  resourceVersion: "8282386"
  uid: 44b3f693-b423-4ee4-bc2f-6ed63d1239f6
spec:
  port:
    targetPort: 8000
  tls:
    termination: edge
  to:
    kind: ""
    name: spi-oauth-service
```
My proposal:
```$ oc delete namespace test; oc create namespace test; oc create route edge -n test --service spi-oauth-service --port 8000 test; timeout 2m bash -x -c -- 'while [ "$(oc get -o jsonpath='{.status.ingress[0].host}' -n test route test)" == "" ]; do sleep 1; done'; oc get -n test -o yaml route test
namespace "test" deleted
namespace/test created
route.route.openshift.io/test created
++ oc get -o 'jsonpath={.status.ingress[0].host}' -n test route test
+ '[' '' == '' ']'
+ sleep 1
++ oc get -o 'jsonpath={.status.ingress[0].host}' -n test route test
+ '[' '' == '' ']'
+ sleep 1
++ oc get -o 'jsonpath={.status.ingress[0].host}' -n test route test
+ '[' '' == '' ']'
+ sleep 1
++ oc get -o 'jsonpath={.status.ingress[0].host}' -n test route test
+ '[' '' == '' ']'
+ sleep 1
++ oc get -o 'jsonpath={.status.ingress[0].host}' -n test route test
+ '[' test-kcp-53t3x7p1e7kx.apps.mkovarik.lab.upshift.rdu2.redhat.com == '' ']'
apiVersion: route.openshift.io/v1
kind: Route
metadata:
  annotations:
    kcp.dev/cluster: root:users:zu:yc:kcp-admin:redhat-appstudio
  creationTimestamp: "2022-10-20T09:06:43Z"
  finalizers:
  - workload.kcp.dev/syncer-cE1LtxcEXYvw2nQIA0qEtQJuYhiFvsguQ0eE5R
  generation: 1
  labels:
    state.workload.kcp.dev/cE1LtxcEXYvw2nQIA0qEtQJuYhiFvsguQ0eE5R: Sync
  name: test
  namespace: test
  resourceVersion: "8314170"
  uid: a94c106d-8de8-4f9b-aad6-46f95d750a95
spec:
  port:
    targetPort: 8000
  tls:
    termination: edge
  to:
    kind: ""
    name: spi-oauth-service
status:
  ingress:
  - conditions:
    - lastTransitionTime: "2022-10-20T09:06:48Z"
      status: "True"
      type: Admitted
    host: test-kcp-53t3x7p1e7kx.apps.mkovarik.lab.upshift.rdu2.redhat.com
    routerCanonicalHostname: router-default.apps.mkovarik.lab.upshift.rdu2.redhat.com
    routerName: default
    wildcardPolicy: None
```

**Review by Michkov — APPROVED** [review]

**Pull request merged**